### PR TITLE
fix(mysql): set proper context for outgoing queries

### PR DESCRIPTION
**Pull request by seemk**
Set the proper context for outgoing mysql queries so `net` and possibly `dns` instrumentation spans are linked correctly.

### CODE REVIEW

**Review by t2t2 — APPROVED** [review]

**Review by mhennoch — APPROVED** [review]

**Review by blumamir — APPROVED** [review]

**Comment by dyladan** [comment]
Can you add a test for it?

**Comment by seemk** [comment]
> Can you add a test for it?

Done, not sure if there is a better way to do it instead of stubbing a private method used in the `query` function. Suggestions welcome :slightly_smiling_face:

File: plugins/node/opentelemetry-instrumentation-mysql/src/instrumentation.ts [hunk 1]
```diff
@@ -328,7 +337,9 @@ export class MySQLInstrumentation extends InstrumentationBase<
           });
         }
         span.end();
-        return originalCallback(...arguments);
+        return context.with(parentContext, () =>
```
Comment (Rauno56) [inline plugins/node/opentelemetry-instrumentation-mysql/src/instrumentation.ts:340]: :point_up:

Comment (seemk) [reply plugins/node/opentelemetry-instrumentation-mysql/src/instrumentation.ts:340]: Both cases produce compiler errors, as the original function is of `Function` type. In the first case it's an overloaded function so unless there's a good/simple way to fix the types I'd leave it as is :thinking:

Comment (Rauno56) [reply plugins/node/opentelemetry-instrumentation-mysql/src/instrumentation.ts:340]: Oh well! Let's not dwell on that too much then!

**Pull request merged**